packages/languages/src/browser/language-client-contribution.ts

```diff
@@ -88,40 +94,43 @@ export abstract class BaseLanguageClientContribution implements LanguageClientCo
         return this.workspace.ready;
     }
 
+    protected readonly toDeactivate = new DisposableCollection();
     activate(): Disposable {
+        if (this.toDeactivate.disposed) {
+            this.doActivate(this.toDeactivate);
+        }
+        return this.toDeactivate;
+    }
+    deactivate(): void {
+        this.toDeactivate.dispose();
+    }
+    protected doActivate(toDeactivate: DisposableCollection): void {
         const options: WebSocketOptions = {};
-        const toDeactivate = new DisposableCollection();
-        toDeactivate.push(Disposable.create(() => {
-            options.reconnecting = false;
-        }));
+        toDeactivate.push(Disposable.create(() => options.reconnecting = false));
         this.connectionProvider.listen({
             path: LanguageContribution.getPath(this),
             onConnection: messageConnection => {
-                if (toDeactivate.disposed) {
+                if (this.toDeactivate.disposed) {
```

Comment:
Q: Why do you use `this.toDeactivate` here and the `toDeactivate` argument above? Maybe I am missing something but wouldn't it be better to either use the argument or remove the argument and use `this.toDeactivate`? Presumably, it has the same effect; I still find it misleading (that's why I have asked).

Reply:
good catch, thanks!